src/services/chain.ts

```diff
@@ -320,7 +320,17 @@ export class ChainContext {
     let lastBlockReceived = lastProcessedBlock;
     provider.on("block", async (blockNumber: number) => {
       try {
+        // Decide if we should process this block before polling for events
+        const shouldProcessBlock =
+          blockNumber % this.processEveryNumBlocks === 0;
+
+        if (!shouldProcessBlock) {
+          log.debug(`Skipping block ${blockNumber}`);
+          return;
```

Comment:
but we can't just skip the block without fetching the events for it...

We can skip the polling, but if we skip the whole block we will miss all orders created in the block 😅